### PR TITLE
Fix --complete flag in orm:ensure-production-settings command

### PR DESCRIPTION
**Pull request by alexander-schranz**
$input->getOption('complete') returns `false` and not `null` if not set.

### CODE REVIEW

File: lib/Doctrine/ORM/Tools/Console/Command/EnsureProductionSettingsCommand.php [hunk 1]
```diff
@@ -62,7 +62,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
         try {
             $em->getConfiguration()->ensureProductionSettings();
 
-            if ($input->getOption('complete') !== null) {
+            if (!$input->getOption('complete')) {
```
Comment (alexander-schranz) [inline lib/Doctrine/ORM/Tools/Console/Command/EnsureProductionSettingsCommand.php:65]: current symfony returns `false` not sure if `null` was returned in older version so I did switch to using `!` instead for checking for false or null.

Comment (SenseException) [reply lib/Doctrine/ORM/Tools/Console/Command/EnsureProductionSettingsCommand.php]: There is a default value that can be set in `addOption`. Without an explicit value it is `null`.

https://github.com/symfony/console/blob/5.x/Command/Command.php#L400

It seems to be the same in 4.0 and 3.0.

Comment (alexander-schranz) [reply lib/Doctrine/ORM/Tools/Console/Command/EnsureProductionSettingsCommand.php]: Seems like its not possible to set a default value for InputOption::VALUE_NONE: https://github.com/symfony/console/blob/356c70659d6b48595f9f275cd7c4de0d5544c49c/Input/InputOption.php#L168

Comment (alexander-schranz) [reply lib/Doctrine/ORM/Tools/Console/Command/EnsureProductionSettingsCommand.php]: So I could go with:

```
$input->getOption('complete') === true
```

```
$input->getOption('complete') !== false
```

or the current version.

The value of the input is set here to `true`:
https://github.com/symfony/console/blob/356c70659d6b48595f9f275cd7c4de0d5544c49c/Input/ArgvInput.php#L248

And default value to `false`:

https://github.com/symfony/console/blob/356c70659d6b48595f9f275cd7c4de0d5544c49c/Input/InputOption.php#L182

Comment (SenseException) [reply lib/Doctrine/ORM/Tools/Console/Command/EnsureProductionSettingsCommand.php]: If it's a boolean then your current syntax should fit. You currently let it step into the if-block when the value is `false`. It also looks like there are no tests yet that covers this class. Can you please create one that covers your changes?

Comment (alexander-schranz) [reply lib/Doctrine/ORM/Tools/Console/Command/EnsureProductionSettingsCommand.php]: @SenseException created the missing test class hope it matches how tests are created in this repository.

**Comment by SenseException** [comment]
@doctrine/team-orm Do we need `OrmFunctionalTestCase` in the test class?

**Comment by ostrolucky** [comment]
I don't think so.

**Comment by alexander-schranz** [comment]
Change to the DoctrineTestCase

**Review by SenseException — APPROVED** [review]

**Comment by alexander-schranz** [comment]
Anything I can do to get this merged?

**Comment by SenseException** [comment]
Currently not. This will be handled by another reviewer.

**Review by ostrolucky — APPROVED** [review]

**Comment by alexander-schranz** [comment]
@SenseException @ostrolucky thx!

**Pull request merged**